# Project: Sample Superstore

## Table of Contents
<ul>
<li><a href="#intro">Introduction</a></li>
<li><a href="#wrangling">Data Wrangling</a></li>
    <ul>
    <li><a href="#ass">Data Assessing</a></li>
    <li><a href="#cln">Data Cleaning</a></li>
    </ul>
<li><a href="#eda">Data Exploratory</a></li>
<li><a href="#conc">Conclusions</a></li>
</ul>

<a id='intro'></a>
## Introduction

> ### Dataset Description
> This dataset shows the shipped goods to USA superstore, collects information from 9994 item, Each row represents a item, each column contains item’s attributes

> **We will try to find out the weak areas where we can work to make more profit and derive all business problems**

>### columns description
>- Ship Mode : Mode of shipping used for shipment delivery
>- Segment : Customer segment product was shipped to
>- Country : Country in which the shipment was delivered
>- City : City in which shipment was delivered
>- State : State in which the shipment was delivered
>- Postal Code : Postal code the shipment was delivered to
>- Region : Country region
>- Category : The category product belongs to
>- Sub-Category : Sub-category of the product
>- Sales : Sale made in USD
>- Quantity : Product quantity
>- Discount : Discount given on the product
>- Profit : Profit/loss made on the sale

In [1]:
# Use this cell to set up import statements for all of the packages that you
#   plan to use.
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
%matplotlib inline

<a id='wrangling'></a>
## Data Wrangling
> In this section, We will load in the data, check for cleanliness, and then trim and clean the dataset for analysis.

<a id='ass'></a>
> ### Data Assessing

In [2]:
df = pd.read_csv('SampleSuperstore.csv')
df.head()

Ship Mode    Segment        Country             City       State  \
0    Second Class   Consumer  United States        Henderson    Kentucky   
1    Second Class   Consumer  United States        Henderson    Kentucky   
2    Second Class  Corporate  United States      Los Angeles  California   
3  Standard Class   Consumer  United States  Fort Lauderdale     Florida   
4  Standard Class   Consumer  United States  Fort Lauderdale     Florida   

   Postal Code Region         Category Sub-Category     Sales  Quantity  \
0        42420  South        Furniture    Bookcases  261.9600         2   
1        42420  South        Furniture       Chairs  731.9400         3   
2        90036   West  Office Supplies       Labels   14.6200         2   
3        33311  South        Furniture       Tables  957.5775         5   
4        33311  South  Office Supplies      Storage   22.3680         2   

   Discount    Profit  
0      0.00   41.9136  
1      0.00  219.5820  
2      0.00    6.8714  
3      0.45 -383.0310  
4      0.20    2.5164

In [3]:
df.shape

(9994, 13)

> First, let's inspect data type and looking for missing data.

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9994 entries, 0 to 9993
Data columns (total 13 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Ship Mode     9994 non-null   object 
 1   Segment       9994 non-null   object 
 2   Country       9994 non-null   object 
 3   City          9994 non-null   object 
 4   State         9994 non-null   object 
 5   Postal Code   9994 non-null   int64  
 6   Region        9994 non-null   object 
 7   Category      9994 non-null   object 
 8   Sub-Category  9994 non-null   object 
 9   Sales         9994 non-null   float64
 10  Quantity      9994 non-null   int64  
 11  Discount      9994 non-null   float64
 12  Profit        9994 non-null   float64
dtypes: float64(3), int64(2), object(8)
memory usage: 1015.1+ KB


> We don't have missing values and we don't need to change the columns data type.

> Let's check if there are any duplicates in the data.

In [5]:
df.duplicated().sum()

17

> we have 17 identical duplivate row in the data, we will need to remove them.

> Now let's cheak if we have any Outliers or mistakes in our columns

In [6]:
df.describe()

Postal Code         Sales     Quantity     Discount       Profit
count   9994.000000   9994.000000  9994.000000  9994.000000  9994.000000
mean   55190.379428    229.858001     3.789574     0.156203    28.656896
std    32063.693350    623.245101     2.225110     0.206452   234.260108
min     1040.000000      0.444000     1.000000     0.000000 -6599.978000
25%    23223.000000     17.280000     2.000000     0.000000     1.728750
50%    56430.500000     54.490000     3.000000     0.200000     8.666500
75%    90008.000000    209.940000     5.000000     0.200000    29.364000
max    99301.000000  22638.480000    14.000000     0.800000  8399.976000

>+ The average Sales is 229.86.
>+ The Sales is range from 0.44 to 22638.48, May be we have outliers in this column.
>+ The average Quantity is 3.79.
>+ The Quantity is range from 1 to 14.
>+ The average Discount is 3.79.
>+ The Discounts is range from 0 to 0.8.
>+ The average Profit is 28.66.
>+ The Profit is range from -6599.98 to 8399.97.

In [7]:
df['Ship Mode'].value_counts()

Standard Class    5968
Second Class      1945
First Class       1538
Same Day           543
Name: Ship Mode, dtype: int64

In [8]:
df['Segment'].value_counts()

Consumer       5191
Corporate      3020
Home Office    1783
Name: Segment, dtype: int64

In [9]:
df['Country'].value_counts()

United States    9994
Name: Country, dtype: int64

In [10]:
df['City'].value_counts()

New York City      915
Los Angeles        747
Philadelphia       537
San Francisco      510
Seattle            428
                  ... 
Glenview             1
Missouri City        1
Rochester Hills      1
Palatine             1
Manhattan            1
Name: City, Length: 531, dtype: int64

In [11]:
df['State'].value_counts()

California              2001
New York                1128
Texas                    985
Pennsylvania             587
Washington               506
Illinois                 492
Ohio                     469
Florida                  383
Michigan                 255
North Carolina           249
Arizona                  224
Virginia                 224
Georgia                  184
Tennessee                183
Colorado                 182
Indiana                  149
Kentucky                 139
Massachusetts            135
New Jersey               130
Oregon                   124
Wisconsin                110
Maryland                 105
Delaware                  96
Minnesota                 89
Connecticut               82
Oklahoma                  66
Missouri                  66
Alabama                   61
Arkansas                  60
Rhode Island              56
Utah                      53
Mississippi               53
Louisiana                 42
South Carolina            42
Nevada        

In [12]:
df['Region'].value_counts()

West       3203
East       2848
Central    2323
South      1620
Name: Region, dtype: int64

In [13]:
df['Category'].value_counts()

Office Supplies    6026
Furniture          2121
Technology         1847
Name: Category, dtype: int64

In [14]:
df['Sub-Category'].value_counts()

Binders        1523
Paper          1370
Furnishings     957
Phones          889
Storage         846
Art             796
Accessories     775
Chairs          617
Appliances      466
Labels          364
Tables          319
Envelopes       254
Bookcases       228
Fasteners       217
Supplies        190
Machines        115
Copiers          68
Name: Sub-Category, dtype: int64

> We don't have input spelling mistakes in the dataset, But we need more investigation in the Sales column to make sure that we don't have outliers.

In [15]:
mean = df['Sales'].mean()
Q3 = (df[df['Sales'] > mean])['Sales'].mean()
df[df['Sales'] > Q3].count()

Ship Mode       662
Segment         662
Country         662
City            662
State           662
Postal Code     662
Region          662
Category        662
Sub-Category    662
Sales           662
Quantity        662
Discount        662
Profit          662
dtype: int64

> There are 662 person buy with more than 75 persent of the sales. So, We don't gave outliers in this column

<a id='cln'></a>
> ### Data Cleaning

> We only need to remove the duplicates rows.

In [16]:
df.drop_duplicates(inplace = True)
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 9977 entries, 0 to 9993
Data columns (total 13 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Ship Mode     9977 non-null   object 
 1   Segment       9977 non-null   object 
 2   Country       9977 non-null   object 
 3   City          9977 non-null   object 
 4   State         9977 non-null   object 
 5   Postal Code   9977 non-null   int64  
 6   Region        9977 non-null   object 
 7   Category      9977 non-null   object 
 8   Sub-Category  9977 non-null   object 
 9   Sales         9977 non-null   float64
 10  Quantity      9977 non-null   int64  
 11  Discount      9977 non-null   float64
 12  Profit        9977 non-null   float64
dtypes: float64(3), int64(2), object(8)
memory usage: 1.1+ MB


<a id='eda'></a>
# Data Exploratory

> Now, we will explort the data to visualize it in **Tableau**

In [17]:
df.to_excel (r'C:\Users\Mo\Desktop\SampleSuperstore\SampleSuperstore_cleaned.xlsx', index = False, header=True)

> You can access the dashboard and interact with it from the link below :
> https://public.tableau.com/views/SampleSuperstore_16510596015010/Dashboard1?:language=en-US&publish=yes&:display_count=n&:origin=viz_share_link

> From the Dashboard we find that :

>+ The Total Sales for all regions is \\$2,296,196. 
>+ The Total Profit is \\$286,241. 
>+ The Total Orders is 37,820. 
>+ The Total Discounts is \\$1,559.19.
>+ The Top 5 Sub-Categorize in USA in the order is Phones-Chairs-Storage-Tables-Binders.
>+ The States with negative (-) total profit in the order are Texas, Ohio, Pennsylvania, Illinois, North Carolina, Colorado, Tennessee, Arizona, Florida and Oregon.

>+ The west region is the best Sales region with  \\$108,330 Total Profit, \\$725,256 Total Sales, 12,234 Total Orders and \\$350 Total Discounts.
>+ California in west region is the best profit state with \\$76,331 Total Profit, \\$457,576 Total Sales, 7,649 Total Orders and \\$145.6 Total Discounts.
>+ In California, Los Angeles is the best profit city with \\$30,431 Total Profit, \\$175,832 Total Sales, 2,876 Total Orders and \\$55.5 Total Discounts.
>+ The most of the orders in Los Angeles dilever with Standard Class Ship Mode.
>+ The Top 5 Sub-Categorize in Los Angeles in the order is Phones-Tables-Storage-Accessories-Chairs.

>+ The south region is the worst Sales region with \\$46,749 Total Profit, \\$391,722 Total Sales, 6,209 Total Orders and \\$238.55 Total Discounts.
>+ Texas in the central region is the worst profit state with \\$-25,751 Total Profit, \\$170,125 Total Sales, 3,718 Total Orders and \\$364.24 Total Discounts.
>+ In Texas, Houston is the worst profit city with \\$-10,175 Total Profit, \\$64,441 Total Sales, 1,460 Total Orders and \\$142.74 Total Discounts.
>+ The most of the orders in Houston dilever with Standard Class Ship Mode.
>+ The Top 5 Sub-Categorize in Houston in the order is Phones-Chairs-Bookcases-Machines-Storage.

<a id='conc'></a>
# Conclusions

> We have explore and visualize the superstore dataset and discover that:
>+ The sales team need to focus on selling more products in **South region**.
>+ The states we need to work to make more profit from them are **Texas**, **Ohio**, **Pennsylvania**, **Illinois**, **North Carolina**, **Colorado**, **Tennessee**, **Arizona**, **Florida** and **Oregon**.
>+ The marketing team in the south region need to focus in promoting the products that will make more profit for us and maybe they can get benefit from the west region marketing plan.
>+ We need to minimize the number of orders that dilever with Same Day ship mode, First Class ship mode and Second class ship mode for all regions.
>+ We need to minimize the discounds and focus on when and in which orders we make it.